lib/tasks/schools/match_establishments.rake

```diff
@@ -0,0 +1,16 @@
+namespace :school do
+  desc 'Update establishment_id column for all active Schools in England and Wales'
+  task :match_establishments, [:path] => :environment do |_t, _args|
+    stats = { perfect: 0, la_plus_en: 0, unmatched: 0 }
+    update = {}
+
+    School.active.where(country: [0, 2]).find_each do |sch| # Only do active schools in England and Wales
+      update[sch.id] = { establishment_id: Lists::Establishment.find_establishment_id_for_school(sch, stats) }
```

Comment:
Rather than just return the ids, you can return the Establishment. When Rails does the update it will automatically use the id of the object. So suggest changing the Establishment method and the code here.